libp2p/Host.cpp

```diff
@@ -145,70 +141,76 @@ void Host::stop()
 
 void Host::doneWorking()
 {
-    // reset ioservice (cancels all timers and allows manually polling network, below)
-    m_ioService.reset();
-
-    DEV_GUARDED(x_timers)
-        m_timers.clear();
-    
-    // shutdown acceptor
-    m_tcp4Acceptor.cancel();
-    if (m_tcp4Acceptor.is_open())
-        m_tcp4Acceptor.close();
-
-    // There maybe an incoming connection which started but hasn't finished.
-    // Wait for acceptor to end itself instead of assuming it's complete.
-    // This helps ensure a peer isn't stopped at the same time it's starting
-    // and that socket for pending connection is closed.
-    while (m_accepting)
-        m_ioService.poll();
-
-    // stop capabilities (eth: stops syncing or block/tx broadcast)
-    for (auto const& h: m_capabilities)
-        h.second->onStopping();
-
-    // disconnect pending handshake, before peers, as a handshake may create a peer
-    for (unsigned n = 0;; n = 0)
-    {
-        DEV_GUARDED(x_connecting)
-            for (auto const& i: m_connecting)
-                if (auto h = i.lock())
-                {
-                    h->cancel();
-                    n++;
-                }
-        if (!n)
-            break;
-        m_ioService.poll();
-    }
-    
-    // disconnect peers
-    for (unsigned n = 0;; n = 0)
+    if (m_capabilities.size())
```

Comment:
Is this really needed? It seems we can leave `doneWorking` as it is, and not having capabilities / sessions / acceptor it won't do anything incorrect.

Reply:
Good question - I'm not sure what the impact of some of these actions are when performed on objects that haven't been fully initialized (e.g. calling m_tcp4Acceptor.cancel() without first opening the acceptor). I'll take a closer look.

Reply:
I ran a quick test and there doesn't appear to be any functional impact but I do get a warning message now about there not being a valid file handle (or something to that affect). As such, I think it's best to leave the code as is.